// asurascans_scraper/Modules/Details.js
const axios = require("axios");
const cheerio = require("cheerio");
const header = {
  'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
  'User-Agent': 'Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36'
}

const details = async (id) => {
  try {
    const url = `https://asuratoon.com/manga/${id}`
    const res = await axios.get(url, { header })
    const $ = cheerio.load(res.data)
    let image = $('img.wp-post-image').attr('src')
    let title = $('h1.entry-title').text()
    let altTitle1 = $('.wd-full span').text()
    let altTitle2 = $('.wd-full h2').text()
    let description = $('.wd-full p').text().trim()
    
   let genre =  $('.wd-full a').map((i, el) => $(el).text()).get()
    let status = $('.imptdt i').text()
    let type = $('.imptdt a').text()
    
   let posts = $('.fmed span time')
     .map((i, el) => $(el).text())
    
 let urls = $('.eph-num a').map((i, el) =>  $(el).attr('href')).get()
 let ids = urls.map((url) => url.replaceAll(/https:\/\/asuratoon.com\//g, ''))
    const data = {
      image: image,
      title: title,
      alternative_title: [altTitle1, altTitle2],
      description: description,
      genre: genre,
      status: status,
      type: type,
      first_post: posts[0],
      last_post: posts[1],
      chapter_ids: ids.reverse()
    }
    return data;
  } catch(err) {
    return ("An Error Occurred!!")
  }
}


module.exports = details
