// asurascans_scraper/Modules/Popular.js
const axios = require("axios");
const cheerio = require("cheerio");
const header = {
  'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
  'User-Agent': 'Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36'
}


const popular = async () => {
  try {
    const url = "https://asuracomics.com/";
    const res = await axios.get(url, { header });
    const $ = cheerio.load(res.data);

    const data = [];
    let summary = [];
    let title = [];
    let status = [];
    let genre = [];
    

   let img = $('.poster img').map((i, el) => {
      if ($(el).attr('src') !== '') 
        return $(el).attr('src')
    })


    $('p').each((i, el) => {
      let summaries = $(el).text()
      if (summaries !== '') {
        summary.push(summaries)
      }
    })


    $('.ellipsis a').each((i, el) => {
      let titles = $(el).text()
      title.push(titles)
    })


    $('.extra-category').each((i, el) => {
      let genres = $(el).text()
      genre.push(genres)
    })


    $('.director').each((i, el) => {
      let statuses = $(el).text().split(" ")[1]
      status.push(statuses)
    })


   let rating = $('.site-vote span')
     .map((i, el) => $(el).text())


   let author = $('.actor')
     .map((i, el) => {
      $(el).text().split(" ")[1] === "" ? null: $(el).text().split(" ")[1];
      })


   let series =  $('.ellipsis a').map((i, el) => {
$(el).attr('href')
.replaceAll(/https:\/\/asuratoon.com\/manga\//g, "")
   })

    for (let i = 0; i < img.length; i++) {
      let datas = {
        image: img[i],
        title: title[i],
        summary: summary[i],
        genre: genre[i],
        ratings: rating[i],
        status: status[i],
        author: author[i],
        series_id: series[i],
      }
      data.push(datas)
    }
    return data;
  } catch (err) {
    return `An Error Occurred!!`
  }
}
module.exports = popular
