// array-flatten/asurascans_scraper/Modules/Details.js
const axios = require("axios");
const cheerio = require("cheerio");
const header = {
  'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
  'User-Agent': 'Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36'
}

const details = async (url) => {
  try {
    const res = await axios.get(url, { header })
    const $ = cheerio.load(res.data)
    let image = $('img.wp-post-image').attr('src')
    let title = $('h1.entry-title').text()
    let altTitle1 = $('.wd-full span').text()
    let altTitle2 = $('.wd-full h2').text()
    let description = $('.wd-full p').text().trim()
    let genre = []
    $('.wd-full a').each((i, el) => {
      let genres = $(el).text()
      genre.push(genres)
    })
    let status = $('.imptdt i').text()
    let type = $('.imptdt a').text()
    let post = []
    $('.fmed span time').each((i, el) => {
      let postt = $(el).text()
      post.push(postt)
    })
    let urls = []
    $('.eph-num a').each((i, el) => {
      let url = $(el).attr('href')
      urls.push(url)
    })

    const data = {
      image: image,
      title: title,
      alternative_title: [altTitle1, altTitle2],
      description: description,
      genre: genre,
      status: status,
      type: type,
      first_post: post[0],
      last_post: post[1],
      urls: urls.reverse()
    }
    return data;
  } catch {
    return ("An Error Occurred!!")
  }
}


module.exports = details
