// array-flatten/asurascans_scraper/Modules/Popular.js
const axios = require("axios");
const cheerio = require("cheerio");
const header = {
  'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
  'User-Agent': 'Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36'
}


const popular = async () => {
  try {
    const url = "https://asuracomics.com/";
    const res = await axios.get(url, { header });
    const $ = cheerio.load(res.data);

    const data = [];
    let img = [];
    let summary = [];
    let title = [];
    let status = [];
    let genre = [];
    let rating = [];
    let author = [];
    let series = [];

    $('.poster img').each((i, el) => {
      let image = $(el).attr('src')
      if (image !== '') {
        img.push(image)
      }
    })


    $('p').each((i, el) => {
      let summaries = $(el).text()
      if (summaries !== '') {
        summary.push(summaries)
      }
    })


    $('.ellipsis a').each((i, el) => {
      let titles = $(el).text()
      title.push(titles)
    })


    $('.extra-category').each((i, el) => {
      let genres = $(el).text()
      genre.push(genres)
    })


    $('.director').each((i, el) => {
      let statuses = $(el).text().split(" ")[1]
      status.push(statuses)
    })


    $('.site-vote span').each((i, el) => {
      let ratings = $(el).text()
      rating.push(ratings)
    })


    $('.actor').each((i, el) => {
      let authors = $(el).text().split(" ")[1]
      if (authors == '') {
        author.push(null)
      } else {
        author.push(authors)
      }
    })


    $('.ellipsis a').each((i, el) => {
      let serieses = $(el).attr('href')
      series.push(serieses)
    })

    for (let i = 0; i < img.length; i++) {
      let datas = {
        image: img[i],
        title: title[i],
        summary: summary[i],
        genre: genre[i],
        ratings: rating[i],
        status: status[i],
        author: author[i],
        series_link: series[i],
      }
      data.push(datas)
    }
    return data;
  } catch (err) {
    return `An Error Occurred!!`
  }
}
module.exports = popular
